File: Algorithm/src/Algorithm/Java/ThisisCodingTest/Implementation/_2night.java
package Algorithm.Java.ThisisCodingTest.Implementation;

public class _2night {

    public static void main(String[] args) {
        System.out.println(solution("a1"));
    }

    public static int solution(String s){
        int row = Integer.parseInt(s.charAt(1)+"")-1;
        int column = s.charAt(0)- 97;
        int cnt = 0;
        int[][] steps = {{-2,-1}, {-2 ,1}, {-1,2},{-1,-2},{2,1},{2,-1},{1,-2},{1,2}};
        for (int i = 0; i < steps.length; i++) {
            if (row + steps[i][0] >= 0 && row + steps[i][0] <= 7 && column + steps[i][1] >= 0 && column + steps[i][1] <= 7) {
                cnt++;
            }
        }
        return cnt;
    }
}

File: Algorithm/src/Algorithm/Python/ThisisCodingTest/4_Implementation/_2night.py
# 현재 나이트 위치 받기
input_data = input()
row = int((input_data[1]))-1
column = int(ord(input_data[0])-97)
steps = [(-2, -1), (-1, -2), (1, -2), (2, -1), (2, 1), (1, 2), (-1, 2), (-2, 1)]
cnt = 0
for step in steps:
    if 0 <= row + step[0] <= 7 and 0 <= column + step[1] <= 7:
        cnt += 1
print(cnt)